app/src/main/java/org/mozilla/fenix/components/metrics/GleanMetricsService.kt

```diff
@@ -531,12 +527,13 @@ class GleanMetricsService(private val context: Context) : MetricsService {
         // because it calls Google ad APIs that must be called *off* of the main thread.
         // These two things actually happen in parallel, but that should be ok because Glean
         // can handle events being recorded before it's initialized.
-        gleanInitializer = MainScope().launch {
+        context.components.performance.visualCompletenessTaskManager.add {
             Glean.registerPings(Pings)
         }
+
         // setStartupMetrics is not a fast function. It does not need to be done before we can consider
         // ourselves initialized. So, let's do it, well, later.
-        gleanSetStartupMetrics = MainScope().launch {
+        context.components.performance.visualCompletenessTaskManager.add {
```

Comment:
nit: we could consider creating a variable to hold `context.components.performance.visualCompletenessTaskManager` to avoid duplication and make it easier to follow